src/components/shared/map/layers/WayPointMarkerLayer.tsx

```diff
@@ -0,0 +1,127 @@
+import {useContext, useEffect} from 'react'
+import MapContext from '../MapContext'
+import createMarkerLayer from '../../../../utils/map/create-marker-layer'
+import {Draw, Modify} from "ol/interaction";
+import VectorSource from "ol/source/Vector";
+import {Icon, Style} from "ol/style";
+import {Geometry} from "ol/geom";
+import Point from "ol/geom/Point"
+import {Point as GeoJsonPoint} from 'geojson'
+import {Feature} from "ol";
+import MapConfigurationService from "../../../../services/map/map-configuration-service";
+import addLayerFeatures from "../../../../utils/map/add-layer-features";
+import VectorLayer from "ol/layer/Vector";
+import {Layer} from "ol/layer";
+import {DrawEvent} from "ol/interaction/Draw";
+import {ModifyEvent} from "ol/interaction/Modify";
+
+interface WayPointMarkerLayerProps {
+    /** An array of GeoJSON Points that will be mapped to markers. */
+    features: GeoJsonPoint[],
+    type?: string,
+    handleSetMarker?: (coordinates: number[], id: number) => void
+}
+
+/**
+ * Adds a geojsonlayer control to a map.
+ */
+const WayPointMarkerLayer = ({features, type, handleSetMarker}: WayPointMarkerLayerProps) => {
+    const {map} = useContext(MapContext)
+
+    useEffect(() => {
+
+        // if map or features are not set do nothing
+        if (!map || !features) {
+            return
+        }
+
+        const mapConfigurationService = new MapConfigurationService()
+        const maxMarkerCount = mapConfigurationService.getMaxMarkerCount()
+        const startIcon = mapConfigurationService.getStartIcon()
+        const endIcon = mapConfigurationService.getEndIcon()
+
+        const setupMarkerLayer = (): VectorLayer<VectorSource> => {
+            let markerLayer = map!.getAllLayers().find((layer: Layer) => layer.getProperties()['name'] === 'marker_layer') as VectorLayer<VectorSource>
+            let layerExtent: number[]
+
+            if(!markerLayer) {
```

Comment:
Am I understanding that part correctly - we check whether the layer has been added yet, and if it has, we're adding the features to it and if it doesn't, we're creating the layer, right?

Under what circumstances can either of these two cases happen? And why do we need to call `marker:ayer.getSource()?.clear(true)`? I believe that would clear the whole layer and then immediately readd the points?